docs/networking.adoc

```diff
@@ -0,0 +1,766 @@
+fd00:1122:3344:103::/64     fe80::aa40:25ff:fe05:c      UG      2 19512229 cxgbe0
+fd00:1122:3344:104::/64     fd00:1122:3344:104::1       U      22 229164829 underlay0
+fd00:1122:3344:105::/64     fe80::aa40:25ff:fe05:40c    UG      2      47 cxgbe1
+fd00:1122:3344:105::/64     fe80::aa40:25ff:fe05:c      UG      2  134024 cxgbe0
+fd00:1122:3344:106::/64     fe80::aa40:25ff:fe05:40c    UG      2 18678270 cxgbe1
+fd00:1122:3344:106::/64     fe80::aa40:25ff:fe05:c      UG      2 3848371 cxgbe0
+fd00:1122:3344:107::/64     fe80::aa40:25ff:fe05:40c    UG      2 75927564 cxgbe1
+fd00:1122:3344:1::/64       fe80::aa40:25ff:fe05:40c    UG      2      51 cxgbe1
+fd00:1122:3344:1::/64       fe80::aa40:25ff:fe05:c      UG      2    2437 cxgbe0
+fd00:1122:3344:2::/64       fe80::aa40:25ff:fe05:40c    UG      2      38 cxgbe1
+fd00:1122:3344:2::/64       fe80::aa40:25ff:fe05:c      UG      2    2437 cxgbe0
+fd00:1122:3344:3::/64       fe80::aa40:25ff:fe05:40c    UG      2   13358 cxgbe1
+fd00:1122:3344:3::/64       fe80::aa40:25ff:fe05:c      UG      2    2437 cxgbe0
+----
+
+Recall that cxgbe0 and cxgbe1 are connected to separate switches in the rack.  So we're seeing the prefixes for the other sleds in this deployment.  We have two routes to reach each sled: one through each switch.  The gateway is the link-local address _of each switch_ on the corresponding link.  One notable exception: the route for this same sled (`fd00:1122:3344:104::/64`) points to `underlay0`, the GZ's VNIC on the sled's underlay network.  In this way, traffic leaving the GZ (whether it originated in this GZ or arrived from one of the switches) is directed to the sled's underlay network etherstub and from there to the right zone VNIC.
+
+(Questions: Why does 107 only have one route?  What are the `fd00:99::` routes?)
+
+There are similar routes for other sleds' prefixes on the bootstrap network.
+
+So traffic from our Nexus zone (`fd00:1122:3344:104::4`) to the Oximeter zone on another sled (`fd00:1122:3344:102::4`) goes:
+
+* in the Nexus zone: via the default route to gateway `fd00:1122:3344:104::1`, which is the GZ on the same sled
+* in the GZ on the Nexus sled: via the per-sled route for the Oximeter sled (`fd00:1122:3344:102::0/64``) to the switch
+* from the switch to the Oximeter sled GZ
+* in the GZ on the Oximeter sled: via the same-sled route to the `underlay0` VNIC
+* through the etherstub's virtual switch to the Oximeter zone's VNIC
+
+=== Data path on the bootstrap network
+
+Once set up, the bootstrap network works largely the same way as the underlay network:
+
+* each sled has an etherstub that implements the per-sled bootstrap network
+* each sled in the rack has two routes to each other sled: one through each switch
+
+It's simpler than the underlay network because only global zones have addresses on the bootstrap network.  None of the per-zone datalinks, routing tables, and netstack stuff from the underlay network applies to the bootstrap network.
```

Comment:
I believe the switch zones are on the bootstrap network. It's also simpler because the bootstrap network will never leave the rack.

Reply:
I realize I didn't even talk about cross-rack connectivity on the underlay network, but I think that's okay given where we are and where this doc is.